tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/statements/arrays/ArrayMutabilityTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.ballerinalang.test.statements.arrays;
+
+import org.ballerinalang.launcher.util.BAssertUtil;
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.BRunUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.values.BBoolean;
+import org.ballerinalang.model.values.BInteger;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.exceptions.BLangRuntimeException;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Test cases for array mutability.
+ */
+public class ArrayMutabilityTest {
+
+    private CompileResult compileResult;
+    private CompileResult resultNegative;
+
+    @BeforeClass
+    public void setup() {
+        compileResult = BCompileUtil.compile("test-src/statements/arrays/array-mutability.bal");
+        resultNegative = BCompileUtil.compile("test-src/statements/arrays/array-mutability-negative.bal");
+    }
+
+    @Test
+    public void testValidArrayAssignment() {
+        BValue[] returnValues = BRunUtil.invoke(compileResult, "testValidArrayAssignment");
+        Assert.assertTrue(((BBoolean) returnValues[0]).booleanValue(), "Expected value of true but found false");
+        Assert.assertEquals(((BInteger) returnValues[1]).intValue(), 100, "Expected value of 100");
+    }
+
+    @Test(description = "",
```

Comment:
Shall we either remove this empty description or add meaningful description. Check all the methods